substra/sdk/schemas.py

```diff
@@ -183,6 +183,51 @@ class _BaseComputePlanSpec(_Spec, abc.ABC):
     aggregatetuples: Optional[List[ComputePlanAggregatetupleSpec]]
     testtuples: Optional[List[ComputePlanTesttupleSpec]]
 
+    def get_dependency_graph(self):
```

Comment:
What do you think of creating a `_insert_into_graph(tuple_graph, tuple_id, in_model_ids)` function that could then be used by all the if clauses? I think it'd remove a bit of code that's almost duplicated but not really.